### PR TITLE
AddDeck component

### PR DESCRIPTION
**Pull request by VitaliKozhurov**

### CODE REVIEW

File: src/features/decks/ui/addDeck/AddDeck.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  tags: ['autodocs'],
+}
+
+export default meta
+type Story = StoryObj<typeof meta>
+
+const CreateDeckWithHooks = (args: AddDeckProps) => {
+  return (
+    <>
+      <AddDeck variant={args.variant} values={args.values} onSubmit={args.onSubmit} />
```
Comment (DenisKulik) [inline src/features/decks/ui/addDeck/AddDeck.stories.tsx:18]: we may not use React.Fragment in this case

File: src/features/decks/ui/addDeck/addDeckForm/AddDeckForm.module.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+.form {
+  padding: 2.4rem 3.4rem 3.6rem 2.4rem;
+
+  .imageBlock {
+    width: 100%;
+    height: 12rem;
+    margin-bottom: 2.4rem;
+
+    & img {
```
Comment (DenisKulik) [inline src/features/decks/ui/addDeck/addDeckForm/AddDeckForm.module.scss:11]: may not be used &

**Review by DenisKulik — APPROVED** [review]

File: src/features/decks/ui/addDeck/addDeckForm/AddDeckForm.module.scss (hunk 2 above)
Comment (VitaliKozhurov) [reply src/features/decks/ui/addDeck/addDeckForm/AddDeckForm.module.scss]: Done!

File: src/features/decks/ui/addDeck/AddDeck.stories.tsx (hunk 1 above)
Comment (VitaliKozhurov) [reply src/features/decks/ui/addDeck/AddDeck.stories.tsx]: Remove React.Fragment

File: src/features/decks/ui/addDeck/AddDeck.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { action } from '@storybook/addon-actions'
+import type { Meta, StoryObj } from '@storybook/react'
+
+import { AddDeck, AddDeckFormValues, AddDeckProps } from './'
```
Comment (vitalyabedik) [inline src/features/decks/ui/addDeck/AddDeck.stories.tsx:4]: change import path

File: src/features/decks/ui/addDeck/uploader/Uploader.tsx
Comment (vitalyabedik) [inline src/features/decks/ui/addDeck/uploader/Uploader.tsx:1]: maybe it component should move to components/ui ?

Comment (VitaliKozhurov) [reply src/features/decks/ui/addDeck/uploader/Uploader.tsx:1]: Done!

File: src/features/decks/ui/addDeck/AddDeck.stories.tsx (hunk 3 above)
Comment (VitaliKozhurov) [reply src/features/decks/ui/addDeck/AddDeck.stories.tsx]: Done!

File: src/components/ui/uploader/Uploader.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  className?: string
+} & UseControllerProps<T>
+
+export const Uploader = <T extends FieldValues>({
+  name,
+  control,
+  children,
+  className,
+  ...restProps
+}: ImageUploaderProps<T>) => {
```
Comment (vitalyabedik) [inline src/components/ui/uploader/Uploader.tsx:22]: :JSX.Element

File: src/features/decks/ui/manageDeck/deckForm/DeckForm.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  values?: {
+    name: string
+    isPrivate?: boolean
+    cover?: string
+  }
+  onSubmit: (data: DeckFormValues) => void
+  onClose: () => void
+}
+
+export const DeckForm = ({ buttonTitle, values, onSubmit, onClose }: AddDeckFormProps) => {
```
Comment (vitalyabedik) [inline src/features/decks/ui/manageDeck/deckForm/DeckForm.tsx:21]: : JSX.Element

**Review by vitalyabedik — APPROVED** [review]
Good!

**Pull request merged**